### DISCUSSION TITLE
Only some voices fully respect pan commands

### DISCUSSION THREAD

**Discussion by jeffkaufman** [FluidSynth/fluidsynth#1065] [Q&A] [answered]
I'm seeing only some voices respond to pan commands.

I'm running:

```
fluidsynth -g 1.0 -i -C no --server \
	 --audio-driver=alsa \
         -o audio.alsa.device=hw:1,0" \
         /usr/share/sounds/sf2/FluidR3_GM.sf2
```

I'm using hard pans as:

```
[0xb0, // CC
  0x0a // pan
  either 0 or 127]
```

Testing with voices 81, 64, 39, 24, 84, 35, 28, and 18, I get full panning, as I would expect.

On the other hand, testing with voices 0, 26, 75, and 16 I get almost no panning.

Is this expected? Is there a way I can get hard panning to work?

(I'm using fluidsynth live, and want to send some instruments only on the left channel, and others only on the right channel)

**Comment by mawe42** [comment, marked as answer]
I haven't checked in the Soundfont, but my guess is that all presets that work well for you are presets using mono samples. Panning controls are unsuitable with stereo samples for hard-left / hard-right panning. I had the same problem [a few years ago](https://lists.nongnu.org/archive/html/fluid-dev/2018-01/msg00006.html).

As a workaround, there is the balance control (CC 8). It is still not ideal for hard-panning stereo samples, as it only reduces the left or right part volume of a stereo sample. But it's the best you can do with stereo samples, I'm afraid.

The only "proper" way for hard left/right splits is to use instruments/presets with mono samples.

Oh, and please note that the built-in effects (chorus, reverb) take mono inputs and generate a stereo output. So for perfect channel separation, you need to either disable the internal effects or use a multi-channel output setup (using jack, for example).

**Comment by jeffkaufman** [comment]
Thanks! I've switched my code to apply CC_8 along with CC_10 every time a hard pan, and it works great!

(I don't need the internal effects, so that's fine)

**Reply by mrbumpy409** [reply]
You can also fix this in the SoundFont as follows:

1. Open the SoundFont in Polyphone.
2. Navigate to the instrument containing the stereo samples.
3. With the instrument selected (global zone), go to Toolbox -> "Override a default modulator".
4. Select default modulator 5 (CC-010 Pan position) and click "Override". This modulator should now appear in the modulator pane at the bottom of the main window.
5. Change the amount value from 1000 to 2000.

Now the panning behavior of that instrument should be identical to a mono instrument.
